### PR TITLE
CodingCalendarViewmodel unit test

### LINKED ISSUE #178: CodingCalendarViewModel unit test
**Expected Behaviour**

Write ViewModel tests for CodingCalendarViewModel

**Parent issue**
#60

### PR DESCRIPTION
**Pull request by jguerrerope**
Fixes #178
**Changes:**

- change RateLimiter to provide read and write methods and to perform better unit testing
- added coding calendar viewmodel unit test

### CODE REVIEW

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java [hunk 1]
```diff
@@ -30,12 +31,14 @@ DatabaseReference providesFirebaseDatabaseReference() {
...
-    @Singleton @Provides
+    @Singleton
+    @Provides
     public UserDao providesUserDao(IITPConnectDatabase iitpConnectDatabase){
         return iitpConnectDatabase.userDao();
     }
 
-    @Singleton @Provides
+    @Singleton
+    @Provides
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java]: The changes in this file are unrelated to the PR. Please don't make unrelated changes. It makes it difficult to find the source of bugs later on.

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/NetworkModule.java [hunk 2]
```diff
@@ -115,7 +115,6 @@ Interceptor authenticator(AuthHolder authHolder) {
     @Singleton
     Cache providesCache() {
         int cacheSize = 10 * 1024 * 1024; // 10 MB
-
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/NetworkModule.java]: Unrelated change.

**Review by dr0pdb — CHANGES_REQUESTED** [review]
Remove unrelated changes. Rest looks good.

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java (hunk 1 above)
Comment (jguerrerope) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java]: using the android/java community standard related class format make your repository more easy to read and contribute. But deal with you 👍

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/NetworkModule.java (hunk 2 above)
Comment (jguerrerope) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/NetworkModule.java]: the same as before

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java (hunk 1 above)
Comment (dr0pdb) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java]: You can create a new issue for and do it. Please revert these unnecessary changes in this PR.

Comment (srv-twry) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java]: You can create a new issue for and do it. Please revert these unnecessary changes in this PR.

Comment (srv-twry) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/DaoModule.java]: You can create a new issue for and do it. Please revert these unnecessary changes in this PR.

**Comment by jguerrerope** [comment]
updated.

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**